packages/clarity-js/src/layout/mutation.ts

```diff
@@ -137,41 +140,62 @@ function handle(m: MutationRecord[]): void {
+  let target = mutation.target;
+  let type = config.throttleDom ? track(mutation, timer, instance, timestamp) : mutation.type;
+  if (type && target && target.ownerDocument) { dom.parse(target.ownerDocument); }
+  if (type && target && target.nodeType == Node.DOCUMENT_FRAGMENT_NODE && (target as ShadowRoot).host) { dom.parse(target as ShadowRoot); }
+  switch (type) {
+    case Constant.Attributes:
+        processNode(target, Source.Attributes, timestamp);
+        break;
+    case Constant.CharacterData:
+        processNode(target, Source.CharacterData, timestamp);
+        break;
+    case Constant.ChildList:
+      processNodeList(mutation.addedNodes, Source.ChildListAdd, timer, timestamp);
+      processNodeList(mutation.removedNodes, Source.ChildListRemove, timer, timestamp);
+      break;
+    case Constant.Throttle:
+      throttledMutations.push({mutation, timestamp});
+      break;
+    default:
+      break;
+  }
+}
 async function process(): Promise<void> {
   let timer: Timer = { id: id(), cost: Metric.LayoutCost };
   task.start(timer);
   while (mutations.length > 0) {
     let record = mutations.shift();
     let instance = time();
     for (let mutation of record.mutations) {
-      let state = task.state(timer);
-      if (state === Task.Wait) { state = await task.suspend(timer); }
-      if (state === Task.Stop) { break; }      
-      let target = mutation.target;
-      let type = config.throttleDom ? track(mutation, timer, instance, record.time) : mutation.type;
-      if (type && target && target.ownerDocument) { dom.parse(target.ownerDocument); }
-      if (type && target && target.nodeType == Node.DOCUMENT_FRAGMENT_NODE && (target as ShadowRoot).host) { dom.parse(target as ShadowRoot); }
-      switch (type) {
-        case Constant.Attributes:
-            processNode(target, Source.Attributes, record.time);
-            break;
-        case Constant.CharacterData:
-            processNode(target, Source.CharacterData, record.time);
-            break;
-        case Constant.ChildList:
-          processNodeList(mutation.addedNodes, Source.ChildListAdd, timer, record.time);
-          processNodeList(mutation.removedNodes, Source.ChildListRemove, timer, record.time);
-          break;
-        case Constant.Suspend:
-          let value = dom.get(target);
-          if (value) { value.metadata.suspend = true; }
-          break;
-        default:
-          break;
-      }
+      processMutation(timer, mutation, instance, record.time)
     }
     await encode(Event.Mutation, timer, record.time);
   }
+
+  let throttledMutationsToProcess = throttledMutations;
+  throttledMutations = [];
+  let processedMutations = false;
+  while (throttledMutationsToProcess.length > 0) {
+    let throttledMutationToProcess = throttledMutationsToProcess.shift();
+    processMutation(timer, throttledMutationToProcess.mutation, time(), throttledMutationToProcess.timestamp);
+    processedMutations = true;
+  }
```

Comment:
Should this part be at the top of the function?
check previously throttled things -> check new things
instead of
check new things -> check previously throttled things

---

main reason: in case order matters, this may add some new things before throttled things that should've been processed
not-so-main reason: every time something is throttled, we immediately check it again

Reply:
I actually want the new things to overwrite the throttled mutations (i improved the logic a bit and made it more clear now thanks for calling this out)